### PR TITLE
[Service Bus Client] Minor Test Refactorings

### PR DESCRIPTION
**Pull request by jsquire**
# Summary

During the recent refactoring efforts to help add determinism and stability to the Service Bus tests, some minor opportunities for further refactoring and cleanup were noted.  

## Goals

- Address some of the minor refactorings that were noted during earlier efforts which were deemed out of scope at the time.

- Continue to adding polish to test cases and infrastructure without increasing risk of failures during nightly test runs.

- Attempt to minimize changes made, keeping the original approach, test cases, code styles.

## Non-Goals

- Refactoring of the general approach for testing, the test cases, code style, and other artifacts of the codebase; changes should be kept as tightly scoped and as minimal as possible.

- Adding of additional test cases or new testing scenarios.

## Details

  - Removing file `q` from the repository root, which was inadvertently added during an earlier pull request.

  - Corrected method signatures for asynchronous tests that had been `async void` to properly reflect a `Task` return type.

  - Marked tests and associated helper methods with explicit access modifiers, to prevent static analysis messages related to unused private members for test methods.

  - Removed a duplicate test collection definition to eliminate test discovery warnings.

  - Refactored test case data sources for `RetryExponentShouldRetryTests` and `RunSerializatTests`, which were previously not serializable, causing test discovery warnings and the inability to enumerate and display test cases accurately in the test runners.

## Last Upstream Rebase

Monday, April 08, 2019 1:48pm (EDT)

## Related and Follow-Up Issues

- [[Service Bus Client] Stabilize Live Tests](https://github.com/Azure/azure-sdk-for-net/issues/5446) (#5446)
- [[Service Bus] Client Library Some Live Tests Have Non-Displayable Data Sources](https://github.com/Azure/azure-sdk-for-net/issues/5718) (#5718)

### CODE REVIEW

File: src/SDKs/ServiceBus/data-plane/tests/Microsoft.Azure.ServiceBus.Tests/AmqpConverterTests.cs [hunk 1]
```diff
@@ -14,8 +14,8 @@ namespace Microsoft.Azure.ServiceBus.UnitTests
     public class AmqpConverterTests
     {
         [Fact]
-        [DisplayTestMethodName]        
-        void Convert_Amqp_message_with_Amqp_value_array_segment_to_SB_message()
+        [DisplayTestMethodName]
```
Comment (pakrym) [inline src/SDKs/ServiceBus/data-plane/tests/Microsoft.Azure.ServiceBus.Tests/AmqpConverterTests.cs]: If I'm not mistaken this attribute can be applied at the class level.

Comment (pakrym) [reply src/SDKs/ServiceBus/data-plane/tests/Microsoft.Azure.ServiceBus.Tests/AmqpConverterTests.cs]: Not sure if it's worth doing now but maybe something to clean up tests in the future.

**Review by pakrym — APPROVED** [review]

File: src/SDKs/ServiceBus/data-plane/tests/Microsoft.Azure.ServiceBus.Tests/AmqpConverterTests.cs (hunk 1 above)
Comment (jsquire) [reply src/SDKs/ServiceBus/data-plane/tests/Microsoft.Azure.ServiceBus.Tests/AmqpConverterTests.cs]: Doesn't look like it.   That's a custom attribute, based on one of the SDK extension points and looks to be allowing method targeting only:

```c#
[AttributeUsage(AttributeTargets.Class | AttributeTargets.Method, AllowMultiple = true, Inherited = true)]
public abstract class BeforeAfterTestAttribute : Attribute
```

Comment (jsquire) [reply src/SDKs/ServiceBus/data-plane/tests/Microsoft.Azure.ServiceBus.Tests/AmqpConverterTests.cs]: Honestly, I'm not sure what value it has at all and had thought about removing it.  It looks like we could hoist to the class level, but all that it is doing is spitting out diagnostic `Begin XXX` and `End XXX` methods to `Console.WriteLine` and `Debug.WriteLine`.    Personally, I think we get more value by just enabling the intrinsic XUnit diagnostics for insight into what the runner is up to.

**Comment by maririos** [comment]
@jsquire I really appreciate the detailed descriptions in your PRs! It is clear what I am looking at, and what is not included.
Thank you!

**Comment by jsquire** [comment]
> @jsquire I really appreciate the detailed descriptions in your PRs! It is clear what I am looking at, and what is not included.
> Thank you!

@maririos:  Thank you.  I appreciate the kind words.  :)

**Pull request merged**